java-rmi/src/com/baeldung/rmi/MessengerServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.baeldung.rmi;
+
+import java.rmi.RemoteException;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+import java.rmi.server.UnicastRemoteObject;
+
+public class MessengerServiceImpl implements MessengerService {
+
+	@Override
+	public String sendMessage(String clientMessage) {
+		
+		String serverMessage = null;
+		if (clientMessage.equals("Client Message")) {
+			serverMessage = "Server Message";
+		}
+		
+		return serverMessage;
+	}
+	
+	public void createStubAndbind() throws RemoteException {
+		
+		MessengerService stub = (MessengerService) UnicastRemoteObject.exportObject((MessengerService) this, 0);
+		Registry registry = LocateRegistry.createRegistry(1099);
+		registry.rebind("MessengerService", stub);
+	}
+	
+	@Override
+	public Message sendMessage(Message clientMessage) throws RemoteException {
+		
+		Message serverMessage = null;
+		if (clientMessage.getMessageText().equals("Client Message")) {
+			serverMessage = new Message("Server Message", "text/plain");
+		}
+		
+		return serverMessage;
+	}
+	
+	public static void main(String[] args) {
+		
+		MessengerServiceImpl msgInf = new MessengerServiceImpl();
+		try {
+			msgInf.createStubAndbind();
+		} catch (RemoteException e) {
+			e.printStackTrace();
```

Comment:
Just my opinion (no action necessary), but the printing of the stack trace happens anyway, right? Not sure why the exception needs to be caught as opposed to putting RemoteException in the throws clause.

Reply:
The intention here is to let reader know that the exception needs to be handled from where the RMI APIs are called. Otherwise the program will crash. This is just an example.

Reply:
Sorry, this was just a temporary code I had written to test it. I will delete the main method from this class.